core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -3910,6 +3910,20 @@ InputPhysicalProperty* NestedJoin::generateIpp(
   InputPhysicalProperty* ipp = NULL;
 
   // ----------------------------------------------------------------
+  // If there is a cardinality constraint on the outer child, we'll
+  // pass that information along to the inner child.
+  // ----------------------------------------------------------------
+  Cardinality minRows;
+  Cardinality maxRows;
+  CardConstraint * outerCardinalityConstraint = NULL;
+
+  if (getGroupAttr()->hasCardConstraint(minRows, maxRows))
```

Comment:
Good catch... I think you are right. I will fix this. I wondered why my unit test succeeded with this mistake. It succeeded because both legs of the join had cardinality constraint of one so the join itself had this constraint also.